src/components/UI/ErrorFallback.js

```diff
@@ -11,18 +14,29 @@ const ErrorWrapperSpan = styled.span`
   font-weight: 300;
 `;
 
-// This function joins error messages into a single string if it receives an array,
-// or returns the error if it is a single value of type string or null.
-// If an array of null values are passed, it returns null.
+/**
+ * Helper that finds the best way to return a pretty-printed
+ * version of the error object passed to it
+ * @param {any} errors - Error object
+ */
 function parseErrors(errors) {
   if (typeof errors === 'string' || !errors) return errors;
 
-  // Remove null values and joins values in a string
-  // TODO maybe we want to show errors in a different manner.
-  const notNullErrors = errors.filter(error => error);
-  if (notNullErrors.length === 0) return null;
+  let message = '';
+
+  if (Array.isArray(errors)) {
+    // Remove nullable values and join values in a string
+    // TODO maybe we want to show errors in a different manner.
+    const notNullErrors = errors.filter(error => error);
+    if (notNullErrors.length === 0) return null;
+
+    message = notNullErrors.join('. ');
+  } else {
+    // Maybe we're lucky and this is an error object/response object
+    message = errors.message || DEFAULT_MESSAGE;
+  }
```

Comment:
propTypes is string or array... don't understand this: my idea was to pass messages (strings) to this component, as this is just responsible to render messages.

Reply:
Good catch about the proptypes.

This addresses objects as well, because there are cases that use objects for errors.

For example:
https://github.com/giantswarm/happa/blob/077c47676ff688532cdcb3de8759165a4443707a/src/actions/nodePoolActions.js#L57

dispatches an error object, and then

https://github.com/giantswarm/happa/blob/077c47676ff688532cdcb3de8759165a4443707a/src/components/Home/ClusterDashboardItem.js#L177
uses the error object, which triggers an error.

Reply:
Yes, my point is that I think we shouldn't be doing this, and if we do we should fix instead letting this component accept objects as props IMHO.

Reply:
Alright, it wasn't clear at first about how it should work, so I thought that was on purpose. I'll fix it

Reply:
Thanks 🙂